packages/pds/src/sql-repo-storage.ts

```diff
@@ -143,205 +150,108 @@ export class SqlRepoStorage extends RepoStorage {
     }
   }
 
-  private getTimestamp(): string {
-    return this.timestamp || new Date().toISOString()
+  async getCarStream(rev?: string) {
+    const root = await this.getRoot()
+    if (!root) {
+      throw new RepoRootNotFoundError()
+    }
+    return writeCarStream(root, async (car) => {
+      let blocks = await this.getBlockRange(500, rev)
+      do {
+        for (const block of blocks) {
+          await car.put(block)
+        }
+        const cidCursor = blocks.at(-1)?.cid
+        if (!cidCursor) {
+          break
+        }
+        blocks = await this.getBlockRange(500, rev, cidCursor)
+      } while (blocks.length > 0)
+    })
   }
 
-  async getCommitPath(
-    latest: CID,
-    earliest: CID | null,
-  ): Promise<CID[] | null> {
-    const res = await this.db.db
-      .withRecursive('ancestor(commit, prev)', (cte) =>
-        cte
-          .selectFrom('repo_commit_history as commit')
-          .select(['commit.commit as commit', 'commit.prev as prev'])
-          .where('commit', '=', latest.toString())
-          .where('creator', '=', this.did)
-          .unionAll(
-            cte
-              .selectFrom('repo_commit_history as commit')
-              .select(['commit.commit as commit', 'commit.prev as prev'])
-              .innerJoin('ancestor', (join) =>
-                join
-                  .onRef('ancestor.prev', '=', 'commit.commit')
-                  .on('commit.creator', '=', this.did),
-              )
-              .if(earliest !== null, (qb) =>
-                // @ts-ignore
-                qb.where('commit.commit', '!=', earliest?.toString() as string),
-              ),
-          ),
-      )
-      .selectFrom('ancestor')
-      .select('commit')
-      .execute()
-    return res.map((row) => CID.parse(row.commit)).reverse()
-  }
+  async getBlockRange(
+    limit: number,
+    rev?: string,
+    cidCursor?: CID,
+  ): Promise<{ cid: CID; bytes: Uint8Array }[]> {
+    let builder = this.db.db
+      .selectFrom('ipld_block')
+      .where('creator', '=', this.did)
+      .orderBy('repoRev', 'asc')
+      .orderBy('cid', 'asc')
+      .select(['cid', 'content'])
+      .limit(limit)
 
-  async getAllBlocksForCommits(commits: CID[]): Promise<BlockForCommit[]> {
-    if (commits.length === 0) return []
-    const commitStrs = commits.map((commit) => commit.toString())
-    const res = await this.db.db
-      .selectFrom('repo_commit_block')
-      .where('repo_commit_block.creator', '=', this.did)
-      .whereRef('repo_commit_block.commit', 'in', valuesList(commitStrs))
-      .innerJoin('ipld_block', (join) =>
-        join
-          .onRef('ipld_block.cid', '=', 'repo_commit_block.block')
-          .onRef('ipld_block.creator', '=', 'repo_commit_block.creator'),
-      )
-      .select([
-        'repo_commit_block.commit',
-        'ipld_block.cid',
-        'ipld_block.content',
-      ])
-      .execute()
+    if (rev) {
+      builder = builder.where('repoRev', '>=', rev)
```

Comment:
At a glance I assumed `rev` would be supplied when you have that revision on hand, and you need to catch-up w/ blocks that come after `rev`.  But I noticed it's inclusive (`>=`) here.  What's the right way to think about the `rev` param?

Reply:
yup you're "at a glance" assumption was correct. this should be an exclusive `>` 👍

Reply:
ah wait i remember why this was the case...

for the initial `since` it's exclusive, but when paginating, it's inclusive

i'll refactor to use a real cursor